### PR TITLE
Fix all circular dependencies

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/next/CONTRIBUTING.md#sending-a-pull-request).

What's done? 
- [x] Remove `import type` 
- [x] Fix all circular dependencies and `no-cycle` eslint rule
- [x] Improve type inference of the wrapper-related props, like `okText`, `PopperProps`, `desktopModeMediaQuery` – make them available only for specific component (side-effect change)
- [x] Enable mocha eslint rules for typescript tests

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -131,6 +131,8 @@ module.exports = {
       files: [
         // matching the pattern of the test runner
         '*.test.js',
+        '*.test.tsx',
+        '*.test.ts',
```
Comment (oliviertassinari) [inline .eslintrc.js:135]: Nice, it sounds like something we should backport to the main repo cc @eps1lon

File: packages/material-ui-lab/src/DatePicker/DatePickerKeyboard.test.tsx [hunk 2]
```diff
@@ -147,7 +147,7 @@ describe('<DatePicker /> keyboard interactions', () => {
       { keyCode: 40, key: 'ArrowDown', expectFocusedDay: 'Aug 20, 2020' },
     ].forEach(({ key, keyCode, expectFocusedDay }) => {
       it(key, () => {
-        fireEvent.keyDown(document.activeElement, { force: true, keyCode, key });
+        fireEvent.keyDown(document.body, { force: true, keyCode, key });
```
Comment (oliviertassinari) [inline packages/material-ui-lab/src/DatePicker/DatePickerKeyboard.test.tsx:150]: On a related note, this makes me think of: @eps1lon shouldn't we refactor the main repo to always fire key events from the active element? To be closer to how the browser works. We rarely do that now.

**Review by oliviertassinari — APPROVED** [review]
Great

File: packages/material-ui-lab/src/DatePicker/DatePickerKeyboard.test.tsx (hunk 2 above)
Comment (eps1lon) [reply packages/material-ui-lab/src/DatePicker/DatePickerKeyboard.test.tsx:150]: This is how I originally thought but later revised it. https://github.com/mui-org/material-ui/pull/20945 has more details.

Comment (oliviertassinari) [reply packages/material-ui-lab/src/DatePicker/DatePickerKeyboard.test.tsx:150]: Oh sorry, I had forgotten about this behavior. Brilliant.

**Pull request merged**